### PR TITLE
[Enhance] introduce fastpath for merge nullable column

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Merge `NullableColumn` needs to be performed in row-wise, which is pretty slow. So we introduce a fastpath for it if it does not contains null datums.

## Experiment Result
As benchmark result, merge `NullableColumn` is much slower than normal column, so this optimization could achieve some benefits in non-null cases.

```
BM_merge_columnwise/2               184215 ns       184190 ns         3670 items_per_second=44.3997M/s
BM_merge_columnwise/6              1110737 ns      1110633 ns          631 items_per_second=21.9929M/s
BM_merge_columnwise/10             2189901 ns      2189480 ns          311 items_per_second=18.6117M/s
BM_merge_columnwise/14             3344204 ns      3343591 ns          206 items_per_second=17.0122M/s
BM_merge_columnwise/18             4569135 ns      4568422 ns          155 items_per_second=16.1032M/s
BM_merge_columnwise/22             5702717 ns      5702183 ns          120 items_per_second=15.6873M/s
BM_merge_columnwise/26             7237213 ns      7236017 ns           97 items_per_second=14.6923M/s
BM_merge_columnwise/30             8318747 ns      8317740 ns           83 items_per_second=14.7444M/s
BM_merge_columnwise/34             9830872 ns      9829731 ns           72 items_per_second=14.095M/s
BM_merge_columnwise/38            11051449 ns     11050333 ns           61 items_per_second=14.051M/s
BM_merge_columnwise/42            12629722 ns     12628535 ns           57 items_per_second=13.6092M/s
BM_merge_columnwise/46            13862942 ns     13861652 ns           47 items_per_second=13.4765M/s
BM_merge_columnwise/50            15780122 ns     15778650 ns           46 items_per_second=12.8908M/s
BM_merge_columnwise/54            17066532 ns     17064833 ns           42 items_per_second=12.8854M/s
BM_merge_columnwise/58            18831904 ns     18830144 ns           39 items_per_second=12.5486M/s
BM_merge_columnwise/62            20746702 ns     20744757 ns           35 items_per_second=12.161M/s
BM_merge_columnwise_nullable/2      367079 ns       367044 ns         1961 items_per_second=22.3188M/s
BM_merge_columnwise_nullable/6     2548951 ns      2548701 ns          278 items_per_second=9.64256M/s
BM_merge_columnwise_nullable/10    5243321 ns      5242831 ns          135 items_per_second=7.81257M/s
BM_merge_columnwise_nullable/14    8074353 ns      8073594 ns           87 items_per_second=7.10266M/s
BM_merge_columnwise_nullable/18    9646131 ns      9645229 ns           59 items_per_second=7.64399M/s
BM_merge_columnwise_nullable/22   14834784 ns     14833397 ns           48 items_per_second=6.07494M/s
BM_merge_columnwise_nullable/26   18250523 ns     18248693 ns           39 items_per_second=5.83582M/s
BM_merge_columnwise_nullable/30   21717895 ns     21715867 ns           33 items_per_second=5.65854M/s
BM_merge_columnwise_nullable/34   25339625 ns     25337265 ns           28 items_per_second=5.49641M/s
BM_merge_columnwise_nullable/38   29536876 ns     29534105 ns           24 items_per_second=5.27011M/s
BM_merge_columnwise_nullable/42   33548381 ns     33545041 ns           21 items_per_second=5.12839M/s
BM_merge_columnwise_nullable/46   37530215 ns     37526710 ns           19 items_per_second=5.02085M/s
BM_merge_columnwise_nullable/50   41891001 ns     41887072 ns           17 items_per_second=4.88934M/s
BM_merge_columnwise_nullable/54   46184711 ns     46180150 ns           15 items_per_second=4.78959M/s
BM_merge_columnwise_nullable/58   50009037 ns     50004334 ns           14 items_per_second=4.75095M/s
BM_merge_columnwise_nullable/62   54388986 ns     54383875 ns           12 items_per_second=4.66962M/s
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: be/src/bench/chunks_sorter_bench.cpp [hunk 1]
```diff
@@ -40,22 +40,25 @@ class ChunkSorterBase {
     void TearDown() { _runtime_state.reset(); }
 
     static std::tuple<ColumnPtr, std::unique_ptr<ColumnRef>> build_sorted_column(TypeDescriptor type_desc,
-                                                                                 int slot_index) {
+                                                                                 int slot_index, bool low_card,
```
Comment (trueeyu) [inline be/src/bench/chunks_sorter_bench.cpp:43]: What's the low_card used for?

**Review by trueeyu — DISMISSED** [review]

**Review by kangkaisen — APPROVED** [review]

File: be/src/bench/chunks_sorter_bench.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/bench/chunks_sorter_bench.cpp:43]: No usage right now.

**Comment by LiShuMing** [comment]
Do you know why nullable columns can be twice slower than non-nullable columns?

**Review by ZiheLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by trueeyu — APPROVED** [review]

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.3 branch-2.4

**Comment by mofeiatwork** [comment]
> 

Cause `NullableColumn` step into the slow path, which is a row-wise comparison with `Column::compare_at`.

**Pull request merged**